tests/unit/client.tcl

```diff
@@ -0,0 +1,78 @@
+start_server {} {
+    test {CLIENT Caching wrong number of arguments} {
+        catch {r client caching} err
+        set _ $err
+    } {ERR*wrong number of arguments*}
+
+    test {CLIENT Caching wrong argument} {
+        catch {r client caching maybe} err
+        set _ $err
+    } {ERR*when the client is in tracking mode*}
```

Comment:
I see some tests all go here `when the client is in tracking mode`, please consider merging them into one (i think the test just need to check the flag, not some `wrong argument` or `without output` etc...)
```
        if (!(c->flags & CLIENT_TRACKING)) {
            addReplyError(c,"CLIENT CACHING can be called only when the "
                            "client is in tracking mode with OPTIN or "
                            "OPTOUT mode enabled");
            return;
        }
```